### PR TITLE
Emit event in case of geolocation error

### LINKED ISSUE #489: Handle errors in geolocation
Errors from ol.Geolocation should be handled

### PR DESCRIPTION
**Pull request by tsauerwein**
Supersedes #560
Closes #489
Closes #560

### CODE REVIEW

**Comment by fgravin** [comment]
Looks good.
But notice that only controllers that are parents (at any level) will receive the event.
For example, a same level directive controller won't catch this event, but this case may never be found.

**Comment by pgiraud** [comment]
We should be able to translate the error message. This can be addressed later though.

File: src/directives/mobilegeolocation.js [hunk 1]
```diff
@@ -12,6 +12,17 @@ goog.require('ol.geom.Point');
 
 
 /**
+ * @enum {string}
+ */
+ngeo.MobileGeolocationEventType = {
+  /**
+   * Triggered when an error occures.
+   *
```
Comment (pgiraud) [inline src/directives/mobilegeolocation.js]: extra line

File: src/directives/desktopgeolocation.js [hunk 2]
```diff
@@ -12,6 +12,18 @@ goog.require('ol.geom.Point');
 
 
 /**
+ * @enum {string}
+ */
+ngeo.DesktopGeolocationEventType = {
+  /**
+   * Triggered when an error occures.
+   *
```
Comment (fredj) [inline src/directives/desktopgeolocation.js]: Extra line

**Comment by fgravin** [comment]
Scope `$on` and `$emit` function are exported angular way to deal with event.
As, internally, we use closure library, i wonder if we should'nt use closure event system ?
It would got over scope tree view issues.

```js
goog.inherits(AbstractController, ol.events.EventTarget);

...

this.dispatchEvent(
```

What do you think ?

**Comment by tsauerwein** [comment]
> As, internally, we use closure library, i wonder if we shouldn't use closure event system ?

This would mean, if you would want to listen to events emitted by the geo-location directive, you would need a reference to the geo-location controller to set up the listener. How practical is that?

As you can install the geo-location directive on any element, I don't think it is a problem that the event is only sent to parents.

**Comment by fgravin** [comment]
Ok you are right

**Comment by fgravin** [comment]
LGTM

**Pull request merged**